pages/bib/[id]/index.tsx

```diff
@@ -122,37 +131,58 @@ export default function BibPage({
       }
     )
     const bibQueryString = getBibQueryString(newQuery, false, viewAllItems)
+    try {
+      // Cancel any active fetches on new ItemTable refreshes
+      if (controllerRef.current) {
+        controllerRef.current.abort()
+      }
+      controllerRef.current = new AbortController()
+      const signal = controllerRef.current.signal
+      const response = await fetch(
+        `${BASE_URL}/api/bib/${bib.id}${bibQueryString}`,
+        {
+          method: "get",
+          signal,
+        }
+      )
+      if (response?.ok) {
+        const { discoveryBibResult } = await response.json()
+        setBib(new Bib(discoveryBibResult))
```

Comment:
Is it not possible to do something like `discoveryBibResult.updateItems(...)` instead of having to create a new bib model and update the state?

You'd have to do it with other properties but it seems better, if possible, in order to reduce one more state variable.

Reply:
Yeah that's possible, the only difference is we'd need to keep track of the variables that need to be refreshed and pass them in explicitly. I'm cool with that approach but pointing it out.

Reply:
That makes sense to me since you're being upfront about what needs to be updated in the model. It's still the same model with the same base information so that's why it doesn't make sense to me to reecreate the model every time. Any thoughts @charmingduchess ?

Reply:
The overhead isn't big, there's really no big risk in re-creating the model, but this is from a conceptual approach.

Reply:
Discussed this offline with Edwin and decided to create a follow-up ticket to optimize state in the component. I added details there:

https://newyorkpubliclibrary.atlassian.net/browse/SCC-4193